File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        
        int n = nums.length;
        
        //sorting 
        Arrays.sort(nums);
        
        //checking with the index
        
        for(int ind =0; ind < n; ind++){
            
            if( ind != nums[ind]){
                return ind;
            }
        }
        
        
        return n;
        
        
    }
}

File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {

            int n = nums.size();
            for(int i=0; i<=n ;i++){
 
              bool ans  = isPresent(i, nums);
 
                if(ans == false){
                    return i;
                }
 
            }
        
        return -1;
    }
    
       bool isPresent(int key, vector<int>& nums){
 
        int n = nums.size();
 
        for(int i=0; i<n;i++){
 
            if(nums[i]  == key ){
                return true;
            }
        }
 
        return false;
 
    }
};